src/banman.h

```diff
@@ -45,10 +46,18 @@ class BanMan
     BanMan(fs::path ban_file, CClientUIInterface* client_interface, int64_t default_ban_time);
     void Ban(const CNetAddr& net_addr, const BanReason& ban_reason, int64_t ban_time_offset = 0, bool since_unix_epoch = false);
     void Ban(const CSubNet& sub_net, const BanReason& ban_reason, int64_t ban_time_offset = 0, bool since_unix_epoch = false);
+    void Discourage(const CNetAddr& net_addr);
     void ClearBanned();
```

Comment:
Looks like ClearBanned() should call `m_discouraged.reset()` otherwise the `clearbanned` RPC no longer does what users expect.

Reply:
I consider that, but I think it would be counterintuitive. Discouragement isn't exposed as bans (it's not visible or accessible in RPCs or UI) as it's a completely automatic process, so I think that exposing a way to clear it would violate that abstraction.

Reply:
I see. The current implementation may introduce subtle bugs then, especially considering that `IsBanned()` is now unintuitively named. `setban "someIP" "add"` may return `Error: IP/Subnet already banned` for instance.

Reply:
Nice catch. I've addressed it by adding a separate `IsBannedOrDiscouraged` that's distinct from `IsBanned`.